go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -650,6 +650,17 @@ func (agent *ActionAgent) Start(ctx context.Context, mysqlHost string, mysqlPort
 // while taking lameduck into account. However, this may be useful for tests,
 // when you want to clean up an agent immediately.
 func (agent *ActionAgent) Stop() {
+	// cleanup initialized fields in the tablet entry
+	f := func(tablet *topodatapb.Tablet) error {
+		tablet.Hostname = ""
+		tablet.MysqlHostname = ""
+		tablet.PortMap = nil
+		return nil
+	}
+	if _, err := agent.TopoServer.UpdateTabletFields(context.Background(), agent.TabletAlias, f); err != nil {
```

Comment:
You need to first check for ownership of the tablet record by this process, same thing as in action_agent.go line 709:
                if err := topotools.CheckOwnership(agent.initialTablet, tablet); err != nil {
                        return err
                }
That makes sure that the current process is still matching the tablet record. If catches the cases where a new tablet was started, but the old one was still running and in shutdown mode. Then the old one cannot mess with the record that has been taken over by the new one.

Reply:
Actually, this code should be in the beginning of the 'f' func, obviously.

Reply:
Oh woops, I actually thought I did that. Must've lost it when I was changing some things around, will re-add.